docs/spec/reactors/block_sync/reactor.md

```diff
@@ -44,10 +44,259 @@ type bcStatusResponseMessage struct {
+
+    upon receiving bcBlockResponseMessage m from peer p:    
+      pool.mtx.Lock()	  
+      requester = pool.requesters[m.Height]	  
+      if requester == nil then	  
+        error("peer sent us a block we didn't expect")	    
+        continue        		
+
+      if requester.block == nil and requester.peerID == p then	  
+        requester.block = m
+        pool.numPending -= 1  // atomic decrement         
+        peer = pool.peers[p]                        
+        if peer != nil then                		
+          peer.numPending--                		  
+          if peer.numPending == 0 then                		  
+            peer.timeout.Stop()          
+            // NOTE: we don't send Quit signal to the corresponding requester task!                  
+        else                 		  
+          trigger peer timeout to expire after peerTimeout          
+      pool.mtx.Unlock()          	  
+		
+		
+    upon receiving bcStatusRequestMessage m from peer p:
+      try to send bcStatusResponseMessage(pool.store.Height)    	
+
+    upon receiving bcStatusResponseMessage m from peer p:
+      pool.mtx.Lock()		
+      peer = pool.peers[p]	  
+      if peer != nil then	  
+        peer.height = m.height	    
+      else	  
+        peer = create new Peer data structure with id = p and height = m.Height	    
+        pool.peers[p] = peer  
+
+      if m.Height > pool.maxPeerHeight then       
+        pool.maxPeerHeight = m.Height		    
+      pool.mtx.Unlock()    
+	    
+onTimeout(p):
+  send error message to pool error channel  
+  peer = pool.peers[p]  
+  peer.didTimeout = true  
+```
+
+### Requester tasks
+
+Requester task is responsible for fetching a single block at position `height`.   
+
+```go
+fetchBlock(height, pool):
+  while true do    
+    peerID = nil	
+    block = nil
+    peer = pickAvailablePeer(height)    
+    peerId = peer.id	
+
+    enqueue BlockRequest(height, peerID) to pool.requestsChannel
+    redo = false    
+    while !redo do	
+      select { 	  
+        upon receiving Quit message do      
+	      return  
+
+	    upon receiving message on redoChannel do  
+	      mtx.Lock()  
+	      pool.numPending++  
+          redo = true        	              
+	      mtx.UnLock()
+      }          
+
+pickAvailablePeer(height):
+  selectedPeer = nil	
+  while selectedPeer = nil do	
+    pool.mtx.Lock()	  
+    for each peer in pool.peers do 		
+      if !peer.didTimeout and peer.numPending < maxPendingRequestsPerPeer and peer.height >= height then		  
+        peer.numPending++		    
+        selectedPeer = peer		    
+        break 
+    pool.mtx.Unlock()                	
+    
+    if selectedPeer = nil then          
+      sleep requestIntervalMS	        		
+
+  return selectedPeer 	
+```
+sleep for requestIntervalMS
+### Task for creating Requesters
+
+This task is responsible for continuously creating and starting Requester tasks.
+```go
+creteRequestors(pool):
```

Comment:
`createRequestors`

Reply:
We need to discuss the formatting issues with .md files. I was fighting yesterday 2 hours to align pseudo-code and it apparently still looks bad. Do we have markdown expert, @zramsay maybe? I am wondering how much formatting support we have with markdown so rendering is more deterministic. Having latex support would be awesome. I will update diagram.

Reply:
We will be implementing a markdown linter in the near future. For now i'd say as long as it is close enough to standard markdown, we're good

Reply:
Thanks @zramsay. I was having issue with formatting, I guess related to tabs and white spaces. The pseudocode is properly formatted in the source file and then after rendering looks bad. I am using Intellij GoLand editor so it could be that it's editor issue. Have you experienced similar issues and do you have some recommendation regarding this?

Reply:
hmm, i use vim and my only plugin is vim-go ... @nylira mentioned using an editor plugin for formatting md files. we could add some info here: https://github.com/tendermint/coding/tree/master/editors

Reply:
We're attempting to use Prettier for markdown formatting of all docs and content. https://github.com/prettier/prettier It's a popular and battle tested code formatter that is integrated with most common text editors.

Reply:
Thanks @zramsay @nylira. I will try prettier on some markdown files.